Makefile

```
@@ -0,0 +1,33 @@
+# Copyright (C) 2018-2020 Ruilin Peng (Nick) <pymumu@gmail.com>.
+#
+# smartdns is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# smartdns is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+.PHONY: all
+
+all: SMARTDNS_BIN
+
+SMARTDNS_BIN : 
+	make -C src all $(MAKEFLAGS)
+
+clean:
+	make -C src clean $(MAKEFLAGS)
+
```

I would suggest renaming the service file to `service.in`, replace the hardcoded paths with placeholders and add the following target to the makefile:

```suggestion

systemd/smartdns.service: systemd/smartdns.service.in
    sed 's|@SBINDIR@|$(SBINDIR)|' $< > $@
    sed 's|@SYSCONFDIR@|$(SYSCONFDIR)|' $< > $@
    sed 's|@RUNSTATEDIR@|$(RUNSTATEDIR)|' $< > $@
```